packages/ui/react-ui-table/src/components/Table/TableBody.tsx

```diff
@@ -0,0 +1,69 @@
+//
+// Copyright 2023 DXOS.org
+//
+
+import { useArrowNavigationGroup } from '@fluentui/react-tabster';
+import { flexRender, type Row, type RowData } from '@tanstack/react-table';
+import React from 'react';
+
+import { useTableContext } from './TableContext';
+import { tbodyTr } from '../../theme';
+import { Cell } from '../Cell/Cell';
+
+type TableBodyProps<TData extends RowData> = {
+  rows: Row<TData>[];
+};
+
+const TABLE_BODY_NAME = 'TableBody';
+
+const TableBody = <TData extends RowData>({ rows }: TableBodyProps<TData>) => {
+  const tableContext = useTableContext<TData>(TABLE_BODY_NAME);
+  const { table, keyAccessor, debug, expand, isGrid, onDatumClick, currentDatum } = tableContext;
+  const rowSelection = table.getState().rowSelection;
+  const domAttributes = useArrowNavigationGroup({ axis: 'grid' });
+  const canBeCurrent = !isGrid && !!onDatumClick;
+  return (
+    <tbody {...(isGrid && domAttributes)}>
+      {rows.map((row) => {
+        const isCurrent = currentDatum === row.original;
+        const isSelected = rowSelection?.[row.id];
+        return (
+          <tr
+            key={keyAccessor ? keyAccessor(row.original) : row.id}
+            className={tbodyTr({ isSelected, isCurrent, canBeCurrent })}
+            {...(isCurrent && { 'aria-current': 'location' })}
+            {...(canBeCurrent && {
+              tabIndex: 0,
+              onClick: () => {
+                onDatumClick(row.original);
+              },
+              onKeyUp: ({ key }) => {
+                if (key === 'Enter' || key === ' ') {
+                  onDatumClick(row.original);
```

Comment:
onRowClick? (and perhaps normalize data => rows in Table?)

Reply:
For `onRowClick` I would anticipate getting a `MouseEvent`, but this calls with `row.original`, hence the unusual name… I’m amenable to renaming this, but I think the name is not a blocker on this approach and would like to discuss it as a follow-up.